build/walk.go

```diff
@@ -23,14 +23,28 @@ package build
 // from outermost to innermost.
 //
 func Walk(v Expr, f func(x Expr, stk []Expr)) {
+	var stack []Expr
+	walk1(&v, &stack, func(x Expr, stk []Expr) Expr {
+		if x != nil {
+			f(x, stk)
+		}
+		return nil
+	})
+}
+
+// Walk2 is similar to Walk1, except that the function is also called with a nil
+// argument after we've visited the node and all its children. This is useful to
+// run code in postorder traversal (the last item of stk contains the current
+// node).
+func Walk2(v Expr, f func(x Expr, stk []Expr)) {
```

Comment:
how about 
WalkIncludingNil

Reply:
WalkIncludingNil doesn't look readable to me.
Renamed to `WalkWithPostfix`. What do you think?